### PR TITLE
NPU Adaption for FLUX

### PR DESCRIPTION
**Pull request by leisuzz**
# What does this PR do?

Implement Flash Attention for NPU machine
Add cleaning memory function for NPU machine

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -1778,7 +1778,28 @@ def __call__(
             query = apply_rotary_emb(query, image_rotary_emb)
             key = apply_rotary_emb(key, image_rotary_emb)
 
-        hidden_states = F.scaled_dot_product_attention(query, key, value, dropout_p=0.0, is_causal=False)
+        if is_torch_npu_available():
```
Comment (yiyixuxu) [inline src/diffusers/models/attention_processor.py:1781]: should we make a new attenton processor instead?

Comment (leisuzz) [reply src/diffusers/models/attention_processor.py]: Thanks for the recommendation, I've modified and make the new attention processor

**Comment by leisuzz** [comment]
@yiyixuxu @sayakpaul Please let me know if there is any improvement for this PR, and thanks for your help!

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
can you run `make style` and `make fix-copies` to the CI would pass?

**Comment by leisuzz** [comment]
@yiyixuxu I've fixed the code quality issue, please approve the workflow. Thanks

**Comment by leisuzz** [comment]
@yiyixuxu Sorry about my mistake, could you please approve the workflow? Thanks

**Comment by yiyixuxu** [comment]
hi @leisuzz 
if you installed our dev environment, you can run `make style` and `make fix-copeis` from the terminal that will automatically adjust the style for you so the CI will pass 

see here https://huggingface.co/docs/diffusers/en/conceptual/contribution#how-to-open-a-pr

**Comment by leisuzz** [comment]
@yiyixuxu Thanks for the information!

**Comment by leisuzz** [comment]
Hi @yiyixuxu ,
Sorry for the trouble, I think there is something wrong with the ruff in my env as there is no code quality in my end. But I think I've fix it manually. Please approve the workflow. Thanks for your help

**Comment by leisuzz** [comment]
@yiyixuxu 
Could you please  redo the approval of the workflow? Thanks for your help and sorry for all the troubles!

**Comment by leisuzz** [comment]
Hi @yiyixuxu ,
There was a copyright / consistency issue with "def fuse_qkv_projections" as it has to match models.unets.unet_2d_condition.UNet2DConditionModel.fuse_qkv_projections. Does that mean I have to change both of these at the same time? For now, please accept this workflows as fuse_qkv_projections can be added later in another PR.

**Comment by leisuzz** [comment]
@yiyixuxu @sayakpaul Could you please approve the workflow? Thanks for your help!

**Comment by sayakpaul** [comment]
Failing tests are unrelated.

**Comment by a-r-r-o-w** [comment]
@yiyixuxu I'm not completely sure why we merged this PR with the following changes:

image

I believe we should default to just the normal SDPA processors otherwise there is inconsistencies introduced. Users can already set custom attention processors, no?

**Comment by yiyixuxu** [comment]
@a-r-r-o-w 
yeah we should revert that - it was my fault, I realized we should not do that here https://github.com/huggingface/diffusers/pull/10465/files

**Pull request merged**